### DISCUSSION TITLE
Valetudo 2023.05.0

### DISCUSSION THREAD

**Discussion by Hypfer** [Hypfer/Valetudo#1829] [Release Discussions]
<div align="center">
    <img src="https://raw.githubusercontent.com/Hypfer/Valetudo/master/assets/logo/valetudo_logo_with_name.svg" width="800" alt="valetudo">
    <p align="center"><h2>2023.05.0</h2></p>
</div>

A few MQTT expansions plus some bugfixes

## MQTT

All the feature changes in this release are related to the MQTT interface:

### Consumables

It is now possible to reset consumables via MQTT.

![image](https://github.com/Hypfer/Valetudo/assets/974410/1e5c1bb0-5b53-4d9e-bfa1-f50c3589af46)

For users of home assistant, this is as easy as clicking on the button entity for the respective consumable that is now also being autoconfigured.
If you're using a different smarthome software, you can find out how to issue the reset command in [the MQTT docs](https://valetudo.cloud/pages/integrations/mqtt.html).

**Note**:
Starting with this release, consumables by default aren't exposed to MQTT anymore, as the information value of them is rather mediocre at best.
This is because at the end of the day, you're looking at linearly decrementing counters that convey a lifetime _assumption_.
A better way to check for the actual consumable state is to just look at the part in question during regular maintenance.

Fear not however, as the functionality is just a checkbox in the MQTT settings away.
If you have an existing deployment, said checkbox will be checked automatically for you during the update process so no breakage here.

### Total statistics

Speaking of optionally exposable capabilities, there's now a new one: Total Statistics

![image](https://github.com/Hypfer/Valetudo/assets/974410/603e2f13-3c02-4a1f-b853-b3a982703f4a)

They work exactly the same as the current statistics but provide all-time values for your automation needs.

### Current statistics

As with the consumables, current statistics are not exposed to MQTT by default anymore in new deployments.

![image](https://github.com/Hypfer/Valetudo/assets/974410/ef53f1bd-b09a-4d84-99c3-75a8be848018)

This is all part of an effort to reduce the amount of data constantly polled from the robot firmware in scenarios where it isn't actually requested or needed by anyone.

### More MQTT buttons

Now that I've figured out that home assistant `button.mqtt` entities exist, I've also added two more to the autodiscovery:

![image](https://github.com/Hypfer/Valetudo/assets/974410/dfc95629-7c4c-4584-ad8d-52f31de20c59)

You can even have ones that are disabled by default for functionality that probably isn't needed by most users:

![image](https://github.com/Hypfer/Valetudo/assets/974410/2dd2d730-5082-4218-9d63-4185981c3382)

Overall these are just a few minor _neat_-ness improvements for the UX.

## Bugfixes

1. Map polling on roborock robots has been improved to fix issues with stale data e.g. when editing virtual restrictions on the S7 Pro Ultra
2. The `mop dock cleaning frequency` quirk for the roborock ultra dock now works correctly
3. When creating multiple timers at once, the editor is now cleared in between to remove stale data

## The usual

If you want to see Valetudo on more robots and/or like this release, you might want to consider donating:

[https://github.com/sponsors/Hypfer](https://github.com/sponsors/Hypfer)

[https://builder.dontvacuum.me/donations.txt](https://builder.dontvacuum.me/donations.txt)

### Features

- **mqtt**: Introduce autoconfigured home assistant buttons for AutoEmpty manual trigger and locate capabilities [`246d655`](http://github.com/Hypfer/Valetudo/commit/246d6558cd3fb5cbd00e5852f14c346c9e2316fa)
- **mqtt**: Allow resetting consumables via MQTT and set consumables to be optionally exposed in new deployments [`ac06012`](http://github.com/Hypfer/Valetudo/commit/ac06012c15c6f282b0b96e8b7081e3355336411a)
- **mqtt**: Optionally expose total statistics and make exposure of current statistics optional for new deployments [`5f85637`](http://github.com/Hypfer/Valetudo/commit/5f85637a773eaf6dac98c2911d61163cbd6996ea)
- **mqtt**: Expose fan speed as its own select entity to home assistant [`4f03d6b`](http://github.com/Hypfer/Valetudo/commit/4f03d6b5e991ae9d3925dd2dd482e706708621c4)
- **ui**: Refresh current statistics every 30s [`63a49a1`](http://github.com/Hypfer/Valetudo/commit/63a49a12de4e9c8a4b74374fdc9b632d0c805d87)
- **vendor.roborock**: Map more error codes [`1b55334`](http://github.com/Hypfer/Valetudo/commit/1b55334dfceaaad30db27f009307190b32863efe)

### Fixes

- **ui**: Hide overflowing status state flag on small screens [`3ccd63f`](http://github.com/Hypfer/Valetudo/commit/3ccd63ffc1073916b81bea5811e7a24202f860da)
- **ui**: Fix timer editor not being reset when creating multiple timers [`342b5e3`](http://github.com/Hypfer/Valetudo/commit/342b5e34fd560ed49a3aae261d2e0646cf6c0144)
- **vendor.dreame**: Handle pixel type 3 [`91003db`](http://github.com/Hypfer/Valetudo/commit/91003dbcfdf61bcfa35c98bcf57c8c9d995c285d)
- **vendor.roborock**: Fix MOP_DOCK_MOP_CLEANING_FREQUENCY quirk [`15a6baa`](http://github.com/Hypfer/Valetudo/commit/15a6baaa8eef54a7c94f239559baba68e6efd59c)
- **vendor.roborock**: Fix virtual restrictions not appearing on refresh on the S7 Pro Ultra + map polling on S5e 1668 [`de9a085`](http://github.com/Hypfer/Valetudo/commit/de9a085a3db9008ad0ec3b64746f168d55ebfd4e)
- Satisfy new sonarcloud rules [`c306028`](http://github.com/Hypfer/Valetudo/commit/c306028cabd8cd40760fd0ba43b40f89b17b37e7)

<hr /><em>This discussion was created from the release <a href='https://github.com/Hypfer/Valetudo/releases/tag/2023.05.0'>Valetudo 2023.05.0</a>.</em>

**Comment by Sylvan86** [comment]
In the image on the "_More MQTT Buttons_" topic, you'll find the entry "_L10s Ultra ThirstySerpentine..._"

Does this mean that Valetudo now runs on the Dreame L10s Ultra or that support for it is in the pipeline?

Otherwise, a big thanks for the work done so far as well as the continuous development.
The changelogs are truly documented in an exemplary manner!

**Reply by Hypfer** [reply]
See https://github.com/Hypfer/Valetudo/discussions/1690#discussioncomment-4710074
